assets/src/edit-story/panels/font.js

```diff
@@ -11,49 +11,66 @@ import { useEffect, useState } from '@wordpress/element';
 /**
  * Internal dependencies
  */
-import { Panel, Title, InputGroup, getCommonValue } from './shared';
+import { useFont } from '../app';
+import { Panel, Title, InputGroup, getCommonValue, SelectMenu } from './shared';
 
 function FontPanel( { selectedElements, onSetProperties } ) {
 	const fontFamily = getCommonValue( selectedElements, 'fontFamily' );
 	const fontSize = getCommonValue( selectedElements, 'fontSize' );
 	const fontWeight = getCommonValue( selectedElements, 'fontWeight' );
 	const fontStyle = getCommonValue( selectedElements, 'fontStyle' );
-	const [ state, setState ] = useState( { fontFamily, fontStyle, fontSize, fontWeight } );
+	const defaultFontWeights = [
+		{ name: 400, slug: 400 },
+	];
+	const { state: { fonts }, actions: { getFont } } = useFont();
+	const [ state, setState ] = useState( { fontFamily, fontStyle, fontSize, fontWeight, fontWeights: defaultFontWeights } );
 	useEffect( () => {
-		setState( { fontFamily, fontStyle, fontSize, fontWeight } );
-	}, [ fontFamily, fontStyle, fontSize, fontWeight ] );
+		const currentFont = getFont( fontFamily );
+		let fontWeights = defaultFontWeights;
+		if ( currentFont ) {
+			const { weights } = currentFont;
+			if ( weights ) {
+				fontWeights = weights.map( ( weight ) => ( { name: weight, slug: weight } ) );
+			}
+		}
+		setState( { fontFamily, fontStyle, fontSize, fontWeight, fontWeights } );
+	}, [ fontFamily, fontStyle, fontSize, fontWeight, getFont ] );
 	const handleSubmit = ( evt ) => {
 		onSetProperties( state );
 		evt.preventDefault();
 	};
+
+	const fontStyles = [
+		{ name: 'normal', slug: 'normal' },
+		{ name: 'italic', slug: 'italic' },
+	];
+
 	return (
 		<Panel onSubmit={ handleSubmit }>
 			<Title>
 				{ 'Font' }
 			</Title>
-			<InputGroup
-				type="text"
+			<SelectMenu
 				label="Font family"
+				options={ fonts }
 				value={ state.fontFamily }
-				isMultiple={ fontFamily === '' }
-				onChange={ ( value ) => setState( { ...state, fontFamily: value } ) }
+				onChange={ ( value ) => {
+					setState( { ...state, fontFamily: value, fontWeight: 400 } );
```

Comment:
Reset back to 400, just in case.

Reply:
Hmm I'd rather not reset if it can be avoided. Can the effect above just check whether the new font supports the existing font weight and only fall back to 400 if it doesn't?

Reply:
Fixed in https://github.com/ampproject/amp-wp/pull/3998/commits/1d13d7e12912b808fdf217ae55bdff3470d95547